### PR TITLE
chore: create bug report form

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Checks

- [x] I have read the [contributing guidelines](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/CONTRIBUTING.md).
- [ ] If some rule is added/updated/removed, I've regenerated the rules list.
- [ ] If some rule meta info is changed, I've regenerated the plugin shared configs.

## Changes

- Create bug report form

## Context

Work in progress.

Go to https://github.com/HonkingGoose/eslint-plugin-testing-library/blob/patch-2/.github/ISSUE_TEMPLATE/bug_report.yml to see a rendered preview of the bug report form.

Helps with issue #400, but doesn't close it as we want forms for other things as well. 😉 

> - Ideally we would have other issue forms to propose a new rule, to request a rule change, or to request a general change.

**GItHub docs for the Issue Form:**

- [Syntax for issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms)
- [Syntax for GitHub's form schema](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema)

### CODE REVIEW

**Comment by HonkingGoose** [comment]
I'm ready for a round of review on this draft. Let me know what you think of it.

I need help with some placeholder text, so if you can suggest a better text, please do so. 😉

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      placeholder: You can find your ESLint configuration in these files TODO ADD LIST OF VALID FILENAMES
+    validations:
+      required: true
+
+  - type: textarea
+    id: rule_affected
+    attributes:
+      label: Rules(s) affected
+      description: Tell us what `eslint-pluging-testing-library` rule(s) are affected by this bug.
+      placeholder: TODO ADD EXAMPLE OF A RULE
```
Comment (HonkingGoose) [inline .github/ISSUE_TEMPLATE/bug_report.yml:94]:
```suggestion
      placeholder: Tip: check your `.eslintrc` for rules.
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      placeholder: 'Tip: you can copy/paste error messages in here. You can click and drag screenshots into this field.'
+    validations:
+      required: false
+
+  - type: textarea
+    id: eslint_config
+    attributes:
+      label: ESLint configuration
+      description: Copy/paste your ESLint configuration into this field.
+      placeholder: You can find your ESLint configuration in these files TODO ADD LIST OF VALID FILENAMES
```
Comment (HonkingGoose) [inline .github/ISSUE_TEMPLATE/bug_report.yml:85]:
```suggestion
      placeholder: Tip: you can find your ESLint configuration in the `.eslintrc` file.
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      label: ESLint configuration
+      description: Copy/paste your ESLint configuration into this field.
+      placeholder: 'Tip: you can find your ESLint configuration in the `.eslintrc` file.'
+    validations:
+      required: true
+
+  - type: textarea
+    id: rule_affected
+    attributes:
+      label: Rules(s) affected
```
Comment (MichaelDeBoey) [inline .github/ISSUE_TEMPLATE/bug_report.yml:92]:
```suggestion
      label: Rule(s) affected
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+      required: true
+
+  - type: textarea
+    id: anything_else
+    attributes:
+      label: Anything else?
+      description: If there's anything else we need to know, tell us about it here.
+      placeholder: By the way, you also need to know about...
+    validations:
+      required: true
```
Comment (MichaelDeBoey) [inline .github/ISSUE_TEMPLATE/bug_report.yml:105]:
```suggestion
      required: false
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

**Review by MichaelDeBoey — APPROVED** [review]

**Review by Belco90 — APPROVED** [review]
It's a really good issue form! I think we got everything covered. Thanks again!

**Comment by HonkingGoose** [comment]
Hooray! Thank you for reviewing and merging my work. 😄

**Pull request merged**